### ISSUE TITLE
baidu.ku6.com no work

### ISSUE THREAD

**Issue by ohofanz** [soimort/you-get#1746]
you-get -d http://baidu.ku6.com/watch/04071100750850531254.html\?page\=videoMultiNeed
you-get: version 0.4.652, a tiny downloader that scrapes the web.
you-get: ['http://baidu.ku6.com/watch/04071100750850531254.html?page=videoMultiNeed']
Traceback (most recent call last):
  File "/usr/local/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1407, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1320, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1136, in download_main
    download(url, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 1400, in any_download
    m.download(url, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/you_get/extractors/ku6.py", line 40, in ku6_download
    ku6_download_by_id(id, output_dir = output_dir, merge = merge, info_only = info_only)
  File "/usr/local/lib/python3.6/site-packages/you_get/extractors/ku6.py", line 11, in ku6_download_by_id
    data = json.loads(get_html('http://v.ku6.com/fetchVideo4Player/%s...html' % id))['data']
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 278, in get_html
    content = get_response(url, faker).data
  File "/usr/local/lib/python3.6/site-packages/you_get/common.py", line 266, in get_response
    response = request.urlopen(url)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

**Comment by ohofanz** [comment]
there is a  redirect url with http code 302 and i can download by the url: 

http://main.gslb.ku6.com/s1/c6W2v7BmGB1Xq3-X/1258467484118/5cef026f72d5d4219e4005b01429d7e6/1489056799479/v596/18/33/zr4eRDAVTxuebIBGdbCSKg.mp4?uuid=AA0DFAE2851B6727B6A26F173D0AA26FD05FFD9D&rate=100&start=10&snqpq=1&act=snapimg&snaptm=10&snapw=80&snaph=60

**Comment by rosynirvana** [comment]
redirect script是这个
```
(function() {
                                    if ( navigator.userAgent.match(/ipad|iphone|android/i) ) {
                        if(window['__setframeinfo__']) window['__setframeinfo__']('url='+encodeURIComponent('http://v.ku6.com/show/VEYbvjS1FEDX18E9.html'));
                        location.href = 'http://v.ku6.com/show/VEYbvjS1FEDX18E9.html';
                    }
```
id 是 VEYbvjS1FEDX18E9

现在的实现会抓到这里
```
src="http://v.baidu.com/watch/04071100750850531254.html/?page\=videoMultiNeed&framehost=baidu.ku6.com&inner"
```
现在这个url不会被重定向到v.ku6.com，所以就拿不到id了

不知道有几种情况

**Comment by soimort** [comment]
Fixed by #1747.
